compiler-core/src/io.rs

```diff
@@ -9,6 +9,21 @@ use tar::{Archive, Entry};
 
 use camino::{Utf8Path, Utf8PathBuf};
 
+/// Takes in a source path and a target path and determines a relative path
+/// from source -> target.
+/// If given a relative target path, no calculation occurs.
+/// # Panics
+/// The provided source path should be absolute, otherwise will panic.
+pub fn make_relative(source_path: &Utf8Path, target_path: &Utf8Path) -> Utf8PathBuf {
+    assert!(source_path.is_absolute());
```

Comment:
Why do we panic in this case?

Reply:
The source path passed in has to be absolute, otherwise the function doesn't work :) So I'd rather panic than continue with an incorrect path.